client/app/bundles/course/material/folders/handles.ts

```diff
@@ -3,21 +3,52 @@ import { getIdFromUnknown } from 'utilities';
 import CourseAPI from 'api/course';
 import { CrumbPath, DataHandle } from 'lib/hooks/router/dynamicNest';
 
+const extractMaterial = (
+  sidebar: { label: string; key: string; path: string }[],
+): { label: string; key: string; path: string } | undefined =>
+  sidebar.find((item) => item.key === 'materials');
+
+export const loadDefaultMaterial = async (
+  courseId: number,
+): Promise<{
+  label: string;
+  key: string;
+  path: string;
+} | null> => {
+  const {
+    data: { sidebar },
+  } = await CourseAPI.courses.fetchLayout(courseId);
+  return sidebar ? extractMaterial(sidebar) || null : null;
+};
```

Comment:
This logic for pulling information from sidebar makes the code convoluted.  

Just open a new `/courses/<course-id>/materials` route and then you can statically reference that route.  
The route can then redirect the user to the root folder.  
This will make the code clean.

Reply:
To be confirmed your idea, this is whats you want ?
![Uploading Screenshot 2024-10-28 at 07.10.13.png…]()

Reply:
Backend needs to open a route as well

Reply:
Have added new backend api and updated the fronted for using it.